compiler/src/iree/compiler/Codegen/Common/TransformDialectJitterPass.cpp

```diff
@@ -328,13 +330,56 @@ static void buildReductionCudaStrategy(ImplicitLocOpBuilder &b,
 
   // Step 5. Post-bufferization mapping to blocks and threads.
   // Need to match again since bufferize invalidated all handles.
-  // TODO: assumes a single func::FuncOp to transform, may ned hardening.
+  // TODO: assumes a single func::FuncOp to transform, may need hardening.
   funcH = b.create<MatchOp>(variantH, func::FuncOp::getOperationName());
   // TODO: extract workgroup_size from config.
-  funcH = mapToBlockAndThreads(b, funcH, ArrayRef<int64_t>{32, 2, 1});
+  funcH = mapToBlockAndThreads(b, funcH, workgroupSize);
 
   // Step 6. Post-bufferization vector distribution with rank-reduction.
   distributeVectors(b, funcH);
+
+  return success();
+}
+
+// TODO: generalize and automate over and over.
+// TODO: significantly shrink this down.
+static LogicalResult buildReductionCpuStrategy(
+    ImplicitLocOpBuilder &b, Value variantH, ArrayRef<int64_t> workgroupSize) {
+  // Step 0. Fetch transform information from the config and materialize it in
+  // the payload IR.
+  // TODO: this still requires specific knowledge of ops present in the IR
+  // and is very brittle.
+  Value originalFillH =
+      b.create<MatchOp>(variantH, linalg::FillOp::getOperationName());
+  Value originalGenericH =
+      b.create<MatchOp>(variantH, linalg::GenericOp::getOperationName());
+  // TODO: no values in config atm.
+  Value tileSizes0Generic =
+      b.create<ConfigExtractPart>(originalGenericH, "tile_sizes", 0);
+  // Value tileSizes1Fill =
+  //     b.create<ConfigExtractPart>(originalFillH, "tile_sizes", 1);
+  // Value tileSizes1Generic =
+  //     b.create<ConfigExtractPart>(originalGenericH, "tile_sizes", 1);
+
+  // Step 1: Distribute to blocks using the current IREE lowering config.
+  variantH = buildReductionStrategyBlockDistributionPart(
+      b, variantH, originalFillH, originalGenericH, tileSizes0Generic);
+
+  // Step 2. Rank-reduce and vectorize.
+  // TODO: assumes a single func::FuncOp to transform, may need hardening.
+  Value funcH = b.create<MatchOp>(variantH, func::FuncOp::getOperationName());
+  funcH = vectorize(b, funcH);
```

Comment:
Naming nit: I'd call this `buildVectorization`, `buildVectorizationStrategy` or something similar to the name of the parent function to avoid giving the impression that it eagerly vectorizes.